rllib/tests/test_catalog.py

```diff
@@ -88,48 +87,90 @@ def test_custom_preprocessor(self):
+        # Build test cases
+        flat_input_case = {
+            "input_space": Box(0, 1, shape=(3,), dtype=np.float32),
+            "output_space": Box(0, 1, shape=(4,)),
+            "num_outputs": 4,
+            "expected_model": "FullyConnectedNetwork",
+        }
+        img_input_case = {
+            "input_space": Box(0, 1, shape=(84, 84, 3), dtype=np.float32),
+            "output_space": Discrete(5),
+            "num_outputs": 5,
+            "expected_model": "VisionNetwork",
+        }
+        flat_complex_input_case = {
+            "input_space": Box(0, 1, shape=(10,), dtype=np.float32),
+            "output_space": Box(0, 1, shape=(5,)),
+            "num_outputs": 5,
+            "expected_model": "FullyConnectedNetwork",
+        }
+        flat_complex_input_case["input_space"].original_space = Tuple([
+            Box(0, 1, shape=(3,), dtype=np.float32),
+            Box(0, 1, shape=(4,), dtype=np.float32),
+            Discrete(3),
+        ])
+        nested_complex_input_case = {
+            "input_space": Tuple([
+                Box(0, 1, shape=(3,), dtype=np.float32),
+                Discrete(3),
+                Tuple([
+                    Box(0, 1, shape=(84, 84, 3), dtype=np.float32),
+                    Box(0, 1, shape=(84, 84, 3), dtype=np.float32),
+                ]),
+            ]),
+            "output_space": Box(0, 1, shape=(7,)),
+            "num_outputs": 7,
+            "expected_model": "ComplexInputNetwork",
+        }
+
+        # Define which tests to run per framework
+        test_suite = {
+            "tf": [
+                flat_input_case,
+                img_input_case,
+                flat_complex_input_case,
+                nested_complex_input_case,
+            ],
+            "tf2": [
+                flat_input_case,
+                img_input_case,
+                flat_complex_input_case,
+                nested_complex_input_case,
+            ],
+            "torch": [
+                flat_input_case,
+                img_input_case,
+                flat_complex_input_case,
+                nested_complex_input_case,
+            ],
+            "jax": [
+                flat_input_case,
+            ],
+        }
+
+        for fw, test_cases in test_suite.items():
+            for test in test_cases:
+                model_config = {}
+                if test["expected_model"] == "ComplexInputNetwork":
+                    model_config["fcnet_hiddens"] = [256, 256]
+                m = ModelCatalog.get_model_v2(
+                    obs_space=test["input_space"],
+                    action_space=test["output_space"],
+                    num_outputs=test["num_outputs"],
+                    model_config=model_config,
+                    framework=fw,
+                )
+                self.assertTrue(test["expected_model"] in type(m).__name__)
+                if isinstance(test["input_space"], Box):
```

Comment:
**todo:** why not do this test for all the obs_spaces? All you need to do is to call `space.sample()` to get an example input, convert to tensor and pass it through the fwd pass?